refactors, adds new features to old zotero import

babslgam opened the pull request.

babslgam committed:
fix: clean whole dataframe

babslgam committed:
feat: ignore zotero attachment items

babslgam committed:
refactor: use correct field name

babslgam committed:
refactor: process edition types

babslgam committed:
refactor: adjust/add fields

babslgam committed:
refactor: use correct field name

babslgam committed:
refactor: remove unneeded date parsing

babslgam committed:
refactor: use correct fields

babslgam committed:
fix: add date via save method (to populate iso field as well)

babslgam committed:
fix: fix typo

babslgam committed:
refactor: use correct fields

babslgam committed:
refactor: only create work if sigle doesn't exist yet

babslgam committed:
refactor: set edtion type

babslgam committed:
refactor: update existing data from zotero

babslgam committed:
refactor: use correct fields for page references

babslgam committed:
feat: parse topics from zotero

babslgam committed:
feat: fill work teaser from expression abstract

babslgam committed:
feat: add new helper function

babslgam committed:
feat: parse and add references between expressions

koeaw commented:
Could I ask that commits which seem to do very similar/same things be combined? E.g. I'm seeing several with identical wording "refactor: use correct fields". Additional commit bodies would additionally help explain what the commit subject means (which I can't guess without looking at the code). 

Also, new features should be split off from refactorings, same for e.g. fixes -> PRs should be focussed on achieving a specific thing + keeping them smaller also makes reviewing them _a lot_ easier.

Thx!

babslgam committed:
feat: sigle is not required anymore

don't depend "importability" of an item on existence of a siglum

babslgam committed:
fix: check for publication date

need to check for pub date because zotero expressions can come without one

babslgam committed:
fix: create a new work if no siglum is provided

get or create based on sigle is not sufficient, so in addition to sigle check we need to create a new work if siglum is empty

koeaw commented:
FYI: I am merging this sight unseen based on new information (to me) re: data imports continuing to be considered very much in flux and subject to change (requests) for another several weeks... which is diametrically opposed to what my assumption was (which drove the expectation of production readiness of submitted code).
